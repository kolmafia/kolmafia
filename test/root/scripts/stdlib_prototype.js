const { print, isTradeable } = require('kolmafia');

const item = Item.get("seal tooth");

const tests = {
    constructor: () => isTradeable.constructor === Function.prototype.constructor,
    call: () => isTradeable.call(null, item) === true,
    apply: () => isTradeable.apply(null, [item]) === true,
    bind: () => {
        const bound = isTradeable.bind(null);
        return bound(item) === true;
    },
    toString: () => isTradeable.toString().length > 0,
    toSource: () => isTradeable.toSource().length > 0,
}

Object.entries(tests).forEach(([k, v]) => print(`${k}: ${v()}`));