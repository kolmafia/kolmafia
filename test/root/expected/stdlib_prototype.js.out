constructor: true
call: true
apply: true
bind: true
toString: true
toSource: true